src/Components/test/E2ETest/ServerExecutionTests/CircuitGracefulTerminationTests.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using BasicTestApp;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests;
+using Microsoft.AspNetCore.E2ETesting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Remote;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.ServerExecutionTests
+{
+    public class CircuitGracefulTerminationTests : BasicTestAppTestBase, IDisposable
+    {
+        public CircuitGracefulTerminationTests(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture.WithServerExecution(), output)
+        {
+        }
+
+        public TaskCompletionSource<object> GracefulDisconnectCompletionSource { get; private set; }
+        public TestSink Sink { get; private set; }
+        public List<(Extensions.Logging.LogLevel level, string eventIdName)> Messages { get; private set; }
```

Comment:
Are we benefiting from having these three as class-level properties? It looks like each of the test cases already takes care of instantiating them and using them independently. Could they just be locals within the test cases?

Reply:
Didn't realize that I left the initializations in there, I'll move them